spec/models/activerecord_obstacle_course_spec.rb

```diff
@@ -274,12 +276,12 @@
 
   it "returns all orders for item_4" do
     # ------------------ Inefficient Solution -------------------
-    order_ids = OrderItem.where(item_id: item_4.id).map(&:order_id)
-    orders = order_ids.map { |id| Order.find(id) }
+    # order_ids = OrderItem.where(item_id: item_4.id).map(&:order_id)
+    # orders = order_ids.map { |id| Order.find(id) }
     # -----------------------------------------------------------
 
     # ------------------ Improved Solution ----------------------
-    #  Solution goes here
+      orders = item_4.orders
```

Comment:
Generally when I see a join I assume many-to-many. Otherwise, I would put a foreign key on one or the other table.

Not sure what you mean when you ask if it would be possible without the `order-item` table, but without that table (everything else the same), there would be no relationship between orders and items, so everything that depended on those relationships would break.